app/controllers/admin/project_steps_controller.rb

```diff
@@ -153,6 +158,7 @@ def render_step_partial(mode)
   def render_modal_content(status = 200)
     @mode = params[:action] == "show" ? :show_and_form : :form_only
     @parents = @step.project.timeline_entries.where(type: "ProjectGroup")
+    @precedents = @step.project.timeline_entries.where("type = 'ProjectStep' AND id != ?", @step.id).by_date
```

Comment:
This array does not work for new steps. We need this functionality to complete this feature.